### PR TITLE
update BlockRequest and BlockResponse to use protobuf encoding

### LINKED ISSUE #1264: encode BlockRequest, BlockResponse messages with protobuf
## Expected Behavior

- In the spec, section 4.2 it outlines how `BlockRequest` and `BlockResponse` messages should be encoded with protobuf
- See [section 4.2 here](https://github.com/w3f/polkadot-spec/blob/master/host-spec/c04-networking.tm#L126)

## Current Behavior

- currently encoded with SCALE

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by noot**
## Changes

- update BlockRequest and BlockResponse messages to use protobuf encoding
- added dot/network/proto files that define the protobuf types
- updated `network.Message` interface `Decode` method to accept `[]byte` (this makes more sense for protobuf)

## Tests

```
go test ./... -short
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- closes #1264

### CODE REVIEW

**Review by RyRy79261 — APPROVED** [review]

File: dot/network/light_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+func TestDecodeLightMessage(t *testing.T) {
+	s := &Service{
+		lightRequest: make(map[peer.ID]struct{}),
+	}
+
+	testPeer := peer.ID("noot")
+
+	testLightRequest := NewLightRequest()
```
Comment (arijitAD) [inline dot/network/light_test.go:18]: Do we have tests for invalid request parsing?

File: dot/network/message.go [hunk 2]
```diff
@@ -71,99 +75,99 @@ type BlockRequestMessage struct {
...
-	// EndBlockHash is an optional type, if next byte is 0 it doesn't exist
-	endBlockHashExists, err := common.ReadByte(r)
-	if err != nil {
-		return err
-	}
-
-	// if endBlockHash was None, then just set Direction and Max
-	if endBlockHashExists == 0 {
-		bm.EndBlockHash = optional.NewHash(false, common.Hash{})
+	if from, ok := msg.FromBlock.(*pb.BlockRequest_Hash); ok {
```
Comment (arijitAD) [inline dot/network/message.go:141]: Replace this with switch case.

File: dot/network/light_test.go (hunk 1 above)
Comment (noot) [reply dot/network/light_test.go:18]: I don't think so, I also didn't write this, I just copied it from service_test.go into a new file

File: dot/network/message.go (hunk 2 above)
Comment (noot) [reply dot/network/message.go]: updated!

**Review by anthdm — APPROVED** [review]
Don't have much insight into the project yet. But looks clean and well written.

**Review by arijitAD — APPROVED** [review]

**Pull request merged**